### DISCUSSION TITLE
using pytest or unittest

### DISCUSSION THREAD

**Discussion by ebrahimraeyat** [xcfem/xc#93] [Ideas]
Hello @lcpt. Did you think about using pytest for test? Because it has many features that we can use, like mock object, test coverage tools and so on. Thanks.

**Comment by ebrahimraeyat** [comment]
I firstly used it on my [pyconcrete](https://github.com/ebrahimraeyat/pyconcrete/tree/master/tests) project.

**Comment by lcpt** [comment]
Hi @ebrahimraeyat 

If I have understood correctly, the main advantage is that you can instruct pytest to perform a series of tests on your code helping to detect errors like the one in `test_shell_mitc9_02.py`. I didn't know this tool, but I agree that it can be very useful.

Thanks.

**Comment by ebrahimraeyat** [comment]
Probably yes. I think it is very useful. There is a real good example of using pytest in ezdxf package, for example look to this file:

https://github.com/mozman/ezdxf/blob/45ef02b98a9fa985e00254e6df16185d82ac1112/tests/test_05_tools/test_517_text_layout.py
